src/main/java/core/basesyntax/operations/BalanceOperation.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class BalanceOperation implements OperationHandler {
+    private final Storage storage;
+
+    public BalanceOperation(Storage storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        if (storage == null) {

```

Comment:
This null check for storage is not needed if you are using dependency injection properly. The instance should be guaranteed to be non-null by the system that provides the dependencies.